tests/responses/test_fileresponse.py

```diff
@@ -80,3 +80,51 @@ def test_unicode_attachment(self):
             response['Content-Disposition'],
             "attachment; filename*=utf-8''%E7%A5%9D%E6%82%A8%E5%B9%B3%E5%AE%89.odt"
         )
+
+    def test_file_to_stream_closes_response(self):
+        # Closing file_to_stream calls FileResponse.close(), even when
+        # file-like object doesn't have a close() method.
+        class FileLike:
+            def read(self):
+                pass
+
+        class FileLikeWithClose(FileLike):
+            def __init__(self):
+                self.closed = False
+
+            def close(self):
+                self.closed = True
+
+        for filelike_cls in (FileLike, FileLikeWithClose):
+            with self.subTest(filelike_cls=filelike_cls.__name__):
+                filelike = filelike_cls()
+                response = FileResponse(filelike)
+                file_to_stream = response.file_to_stream
+                self.assertFalse(response.closed)
+                file_to_stream.close()
+                # Object with close() is added to the list of closable.
+                if hasattr(filelike, 'closed'):
+                    self.assertEqual(response._closable_objects, [filelike])
```

Comment:
I think it's better to test `_closable_objects` immediately after instantiating the `FileResponse` object and before calling `file_to_stream.close()` because `_closable_objects` is set when instantiating the `FileResponse` object. (It's more important to check the state of `_closable_objects` before calling `close()` rather than after because it's used by `close()` and a priori `close()` could change how it was.)